packages/components/table/src/filter-panel.vue

```diff
@@ -203,21 +214,20 @@ export default defineComponent({
       tooltipVisible.value = false
     }
     const handleConfirm = () => {
+      needCached.value = false
       confirmFilter(filteredValue.value)
       hidden()
     }
     const handleReset = () => {
+      needCached.value = false
       filteredValue.value = []
       confirmFilter(filteredValue.value)
       hidden()
     }
     const handleSelect = (_filterValue?: string) => {
       filterValue.value = _filterValue
-      if (!isPropAbsent(_filterValue)) {
-        confirmFilter(filteredValue.value)
-      } else {
-        confirmFilter([])
-      }
+      const value = !isPropAbsent(_filterValue) ? _filterValue : []
+      confirmFilter(value)
```

Comment:
This part doesn't look like it needs to be changed. No need to add an extra variable.